src/metabase/automagic_dashboards/core.clj

```diff
@@ -243,11 +245,15 @@
 (defn- source
   [card]
   (cond
```

Comment:
This is the fundamental logic change -- if the card is a model, x-ray it as a GenericTable. Nearly all other changes are documentation, tests, and refactors for better understanding.